app/src/main/java/org/wikipedia/suggestededits/SuggestedEditsContributionsFragment.kt

```diff
@@ -0,0 +1,487 @@
+    }
+
+    private fun showError(t: Throwable) {
+        swipeRefreshLayout.isRefreshing = false
+        contentContainer.visibility = GONE
+        errorView.setError(t)
+        errorView.visibility = VISIBLE
+    }
+
+    private fun updateFilterViewUI() {
+        val view: SuggestedEditsTypeItem
+        val count: Int
+        when (editFilterType) {
+            EDIT_TYPE_ARTICLE_DESCRIPTION -> {
+                view = articleDescriptionView
+                count = SuggestedEditsUserStats.totalDescriptionEdits
+            }
+            EDIT_TYPE_IMAGE_CAPTION -> {
+                view = imageCaptionsView
+                count = SuggestedEditsUserStats.totalImageCaptionEdits
+            }
+            EDIT_TYPE_IMAGE_TAG -> {
+                view = imageTagsView
+                count = SuggestedEditsUserStats.totalImageTagEdits
+            }
+            else -> {
+                view = allTypesView
+                count = SuggestedEditsUserStats.totalEdits
+            }
+        }
+        contributionsCountText.text = getString(R.string.suggested_edits_contribution_type_title, count, resources.getQuantityString(R.plurals.suggested_edits_contribution, count))
+        for (filterView in filterViews) {
+            if (filterView == view) {
+                filterView.setEnabledStateUI()
+            } else {
+                filterView.setDisabledStateUI()
+            }
+        }
+    }
+
+    private fun loadDataBasedOnFilter() {
+        when (editFilterType) {
+            EDIT_TYPE_ARTICLE_DESCRIPTION -> {
+                articleAndImageContributions.addAll(articleContributions)
+            }
+            EDIT_TYPE_IMAGE_CAPTION -> {
+                articleAndImageContributions.addAll(imageContributions.filter { it.editType == EDIT_TYPE_IMAGE_CAPTION })
+            }
+            EDIT_TYPE_IMAGE_TAG -> {
+                articleAndImageContributions.addAll(imageContributions.filter { it.editType == EDIT_TYPE_IMAGE_TAG })
+            }
+            else -> {
+                articleAndImageContributions.addAll(articleContributions)
+                articleAndImageContributions.addAll(imageContributions)
+            }
+        }
+    }
+
+    private class HeaderViewHolder internal constructor(itemView: View) : DefaultViewHolder<View?>(itemView) {
+        var headerText: TextView = itemView.findViewById(R.id.section_header_text)
+        fun bindItem(date: String) {
+            headerText.text = date
+            headerText.setTextColor(ResourceUtil.getThemedColor(headerText.context, R.attr.colorAccent))
+        }
+    }
+
+    private class ContributionItemHolder internal constructor(itemView: SuggestedEditsContributionsItemView) : DefaultViewHolder<SuggestedEditsContributionsItemView?>(itemView) {
+        val disposables = CompositeDisposable()
+        fun bindItem(contribution: Contribution) {
+            view.setTitle(contribution.description)
+            view.setDescription(contribution.title)
```

Comment:
You're passing `description` into `setTitle()` and `title` into `setDescription`? 🤔

Reply:
Robin wants the title to be the contribution, which is always the description... but I dont want to store it as the "title" in the object as it is confusing.. also, naming the 'title' as 'desc..' on ui is confusing... hence this 😅